content/kb/tutorials/llm/index.md

```diff
@@ -0,0 +1,144 @@
+
+*This tutorial is adapted from a [blog](https://blog.streamlit.io/langchain-tutorial-1-build-an-llm-powered-app-in-18-lines-of-code/) post by Chanin Nantesanamat.*
+
+<Cloud src="https://doc-tutorial-llm-18-lines-of-code.streamlit.app/?embed=true" height="600" />
+
+### Objectives
+
+1. Get an OpenAI key from the end user.
+2. Validate the user's OpenAI key.
+3. Get a text prompt from the user.
+4. Authenticate OpenAI with the user's key.
+5. Send the user's prompt to OpenAI's API.
+6. Get a response and display it.
+
+Bonus: Deploy the app on Streamlit Community Cloud!
+
+### Prerequisites
+
+- Python 3.7+
+- Streamlit
+- LangChain
+- [OpenAI API key](https://platform.openai.com/account/api-keys?ref=blog.streamlit.io)
+
+### Setup coding environment
+
+In your IDE (integrated coding environment), open the terminal and install the following three Python libraries:
+
+```python
+pip install streamlit openai langchain
+```
+
+Create a `requirements.txt` file located in the root of your working directory and save these dependencies. This is necessary for deploying the app to the Streamlit Community Cloud later.
+
+```python
+streamlit
+openai
+langchain
+```
+
+### Building the app
+
+The app is only 18 lines of code:
+
+```python
+import streamlit as st
+from langchain.llms import OpenAI
+
+st.title('🦜🔗 Quickstart App')
+
+openai_api_key = st.sidebar.text_input('OpenAI API Key')
+
+def generate_response(input_text):
+  llm = OpenAI(temperature=0.7, openai_api_key=openai_api_key)
+  st.info(llm(input_text))
+
+with st.form('my_form'):
+  text = st.text_area('Enter text:', 'What are the three key pieces of advice for learning how to code?')
+  submitted = st.form_submit_button('Submit')
+  if not openai_api_key.startswith('sk-'):
+    st.warning('Please enter your OpenAI API key!', icon='⚠')
+  if submitted and openai_api_key.startswith('sk-'):
+    generate_response(text)
+```
+
+To start, create a new Python file and save it as `streamlit_app.py` in the root of your working directory.
+
+1. Import the necessary Python libraries.
+
+    ```python
+    import streamlit as st
+    from langchain.llms import OpenAI
+    ```
+
+2. Create the app's title using `st.title`.
+
+    ```python
+    st.title('🦜🔗 Quickstart App')
+    ```
+
+3. Add a text input box for the user to enter their OpenAI API key.
+
+    ```python
+    openai_api_key = st.sidebar.text_input('OpenAI API Key')
+    ```
+
+4. Define a function to authenticate to OpenAI API with the user's key, send a prompt, and get an AI-generated response. This function accepts the user's prompt as an argument and displays the AI-generated response in a blue box using `st.info`.
+
+    ```python
+    def generate_response(input_text):
+    llm = OpenAI(temperature=0.7, openai_api_key=openai_api_key)
+    st.info(llm(input_text))
```

Comment:
Fix indents

Reply:
Same as below. Not sure how that got changed since it doesn't display like that locally, but I'll confirm.